presto-cli/src/test/java/com/facebook/presto/cli/TestConsoleHistory.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.cli;
+
+import jline.console.history.MemoryHistory;
+import jline.console.history.PersistentHistory;
+import org.testng.annotations.Test;
+
+import java.lang.reflect.Method;
+
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertNotNull;
+
+public class TestConsoleHistory
+{
+    private static Object invokeGetHistory() throws Exception
```

Comment:
Instead of hacks like this, can you please extract `MemoryHistoryProvider` where you would be able to pass path where history is stored. It will make your test much more easier.

Reply:
Instead of extracting a class you can make `Console::getHistory()` package-private and annotate it with `@VisibleForTesting` (which we do in many other places), and then add  a path parameter.